cmd/check.go

```diff
@@ -0,0 +1,246 @@
+package cmd
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+	"regexp"
+	"strconv"
+	"strings"
+	"text/tabwriter"
+
+	"github.com/spf13/cobra"
+)
+
+func init() {
+	rootCmd.AddCommand(checkCmd)
+}
+
+type requirement struct {
+	name    string
+	command string
+	args    []string
+	docsURL string
+	checker func([]byte) (string, error)
+}
+
+type versionError struct {
+	errorText string
+}
+
+type commandError struct {
+	Command    string
+	ErrorText  string
+	Suggestion string
+}
+
+func (e *versionError) Error() string {
+	return fmt.Sprintf("%s", e.errorText)
+}
+
+func (e *commandError) Error() string {
+	return fmt.Sprintf("%s", e.ErrorText)
+}
+
+var checkCmd = &cobra.Command{
+	Use:   "check",
+	Short: "Print the check number of commit0",
+	Run: func(cmd *cobra.Command, args []string) {
+		// Add any new requirements to this slice.
+		required := []requirement{
+			{
+				name:    "AWS CLI\t\t",
+				command: "aws",
+				args:    []string{"--version"},
+				docsURL: "",
+				checker: func(output []byte) (string, error) {
+					ver := ""
+					re := regexp.MustCompile(`aws-cli/([0-9]+)\.([0-9]+)\.([0-9]+)`)
+					m := re.FindStringSubmatch(string(output))
+					major, err := strconv.ParseInt(m[1], 0, 64)
+					if err != nil {
+						return ver, err
+					}
+					minor, err := strconv.ParseInt(m[2], 0, 64)
+					if err != nil {
+						return ver, err
+					}
+					patch, err := strconv.ParseInt(m[3], 0, 64)
+					if err != nil {
+						return ver, err
+					}
+
+					ver = fmt.Sprintf("%d.%d.%d", major, minor, patch)
+
+					if major < 1 || (major == 1 && minor < 16) {
```

Comment:
Can we push these version numbers up to config? Even if it's just hard-coded config for now.

Reply:
Can we reliably count on all the tools we are checking adhering to some version of semver? If so we could have this as part of the config or even use a semver library to do these calculations.

Reply:
My initial thought was to simply have each command use a regex and a semver minimum version as part of the config, then push all the logic to a shared function. But I was concerned that perhaps some tools wouldn't fix that. Given all the current tools allow that I might just go back to it.

Reply:
Looks good. If we introduce a dependency that doesn't use semver we can reassess then.